tests/search.test.ts

```diff
@@ -992,6 +992,25 @@ describe.each([
       client.index(index.uid).search('prince', {}),
     ).rejects.toHaveProperty('cause.code', ErrorStatusCode.INDEX_NOT_FOUND);
   });
+
+  test(`${permission} key: Search with locales`, async () => {
+    const localIndex = (await getClient(permission)).index(index.uid);
+    const localMasterIndex = (await getClient('Master')).index(index.uid);
+
+    const updateLocalizedAttributesEnqueuedTask =
+      await localMasterIndex.updateLocalizedAttributes([
+        { attributePatterns: ['title', 'comment'], locales: ['fra', 'eng'] },
+      ]);
+    await localMasterIndex.waitForTask(
+      updateLocalizedAttributesEnqueuedTask.taskUid,
+    );
+
+    const searchResponse = await localIndex.search('french', {
+      locales: ['fra', 'eng'],
+    });
+
+    expect(searchResponse.hits.length).toEqual(2);
+  });
```

Comment:
Here is a small suggestion for consistency with our other tests 😊

```suggestion
  test(`${permission} key: Search with locales`, async () => {
    const client = await getClient(permission);
    const masterClient = await getClient('Master');

    const { taskUid } = await masterClient
      .index(index.uid)
      .updateLocalizedAttributes([
        { attributePatterns: ['title', 'comment'], locales: ['fra', 'eng'] },
      ]);
    await masterClient.waitForTask(taskUid);

    const searchResponse = await client.index(index.uid).search('french', {
      locales: ['fra', 'eng'],
    });

    expect(searchResponse.hits.length).toEqual(2);
  });
```